monolith_filemanager/file/pandas_file.py

```
@@ -52,18 +52,30 @@ def __init__(self, path: Union[str, FilePath]) -> None:
         """
         super().__init__(path=path)
 
-    def read(self, lazy: bool = False, chunk_size: Union[int, str] = '256MB', **kwargs) -> DataFrameType:
+    def read(self, lazy: bool = False, chunk_size: Union[int, str] = '256MB', reset_index: bool = True,
+             **kwargs) -> DataFrameType:
         """
         Gets data from file defined by file path.
+        If `lazy` is True, then returns a Dask DataFrame.
+        If `lazy` is False, then returns a Pandas DataFrame. The index can be reset to avoid inconsistency with how Dask
+            stores DataFrame indexes.
 
         :param lazy: (bool) Whether reading should be lazy (returns dask DataFrame) or eager (returns pandas DataFrame)
         :param chunk_size: (dask compatible int or str) size in bytes of chunks to read. Only used if lazy == True
-        :return: Data from file
+        :param reset_index: (bool) Whether or not to reset the index of the read dataframe.
+            Only resets the index of pandas dataframes
+        :return: (DataFrameType) Data from file
         """
         storage_options = {'config_kwargs': {'max_pool_connections': 32}, 'skip_instance_cache': True}
-        return self._read_dask(chunk_size, storage_options=storage_options, **kwargs) if lazy \
+        df = self._read_dask(chunk_size, storage_options=storage_options, **kwargs) if lazy \
             else self._read_dask(chunk_size, storage_options=storage_options, **kwargs).compute()
 
+        if reset_index and not lazy:
```

So it seems like you can pass in reset_index=True, and if you are also passing in lazy then it wont reset the index.

I think it might be best to raise an exception in such a situation. Otherwise people might get confused as to why it's not resetting the index.

---

Yeah agreed good to highlight this combination of inputs. I'd maybe put it as a warning level rather than raising an exception -- the code still works if you pass both kwargs as True, but it just means that `reset_index` has no effect. 

Another reason to avoid exception, the default for `reset_index=True` means that any existing usages of `.read(..., lazy=True)` where `reset_index` is not specified will have the "both are true" combination. So if it were an exception, we'd need to find all usages and explicitly put `reset_index=False` which might be a lot of work.

---

@McPonolith would a warning be ok rather than an exception, in particular for the second reason that Will mentioned. All instances of `lazy=True` will require reset_index to be specified as `False`

---

Im not a fan of the warning message route, as theres already existing code thats gonna spit out the warning and be ignored. It would be better to throw an exception cause people would be more motivated to update the code and tests would catch it.

We could rename the attribute maybe `eagerly_reset_index` `eager_reset_index` `reset_index_in_eager`, `if_eager_reset_index`, or something along those lines

Also why does this need to be the default behaviour for eager loading but not lazy loading. Surely the same issue exists when doing lazy?

BTW Im not gonna stand in the way of accepting this PR, so if you're happy the way it is, let me know and I'll accept it.

---

Lazy loading uses Dask DataFrames which doesn't index dataframes in the same way as Pandas. 

I didn't include an option to reset the dask df index because I would be more worried about someone resetting the index of a dask dataframe without knowing that the underlying behaviour isn't what they expect. 

It's also a surprisingly expensive operation with Dask, so will add a big overhead to all Dask manipulators if it's used by default, whereas it's comparatively cheap to do with Pandas

---

"as theres already existing code thats gonna spit out the warning and be ignored" what do you mean by this?

---

Renamed kwarg to `reset_index_if_eager`. Does that provide enough clarity without needing to add an error or warning?